storage/innobase/handler/ha_innopart.cc

```diff
@@ -197,7 +197,14 @@ Ha_innopart_share::open_one_table_part(
 		&& (m_table_share->fields
 		    != dict_table_get_n_user_cols(ib_table)
 		       + dict_table_get_n_v_cols(ib_table) - 1))) {
-		ib::warn() << "Partition `" << get_partition_name(part_id)
+        /* We may end up here when doing ha_innopart::open(), so before
+           populate_partition_name_hash() was called. In such a case
+           get_partition_name() returns nullptr. Fall back to any other
+           informative string instead of trying to print nullptr. */
```

Comment:
```suggestion
        /* We may end up here when doing ha_innopart::open(), so before
        populate_partition_name_hash() was called. In such a case
        get_partition_name() returns nullptr. Fall back to any other
        informative string instead of trying to print nullptr. */
```